code/game/machinery/fusion_engine.dm

```diff
@@ -153,12 +153,12 @@
 					user.visible_message("<span class='notice'>[user] fumbles around figuring out [src]'s internals.</span>",
 					"<span class='notice'>You fumble around figuring out [src]'s internals.</span>")
 					var/fumbling_time = 100 - 20 * user.mind.cm_skills.engineer
-					if(!do_after(user, fumbling_time, TRUE, 5, BUSY_ICON_BUILD))
-						return
+					if(!do_after(user, fumbling_time, TRUE, src, USER_ICON_UNSKILLED, extra_checks = CALLBACK(WT, /obj/item/tool/weldingtool/proc/isOn)))
+						return FALSE
 				playsound(loc, 'sound/items/weldingtool_weld.ogg', 25)
 				user.visible_message("<span class='notice'>[user] starts welding [src]'s internal damage.</span>",
 				"<span class='notice'>You start welding [src]'s internal damage.</span>")
-				if(do_after(user, 200, TRUE, 5, BUSY_ICON_BUILD))
+				if(do_after(user, 200, TRUE, src, USER_ICON_BUILD, extra_checks = CALLBACK(WT, /obj/item/tool/weldingtool/proc/isOn)))
 					if(buildstate != FUSION_ENGINE_HEAVY_DAMAGE || is_on || !WT.isOn())
```

Comment:
Is you're already putting the extra check there you should remove this, or remove the extra check